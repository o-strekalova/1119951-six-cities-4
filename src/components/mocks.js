export const offersAll = [{
  id: `1010`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Amsterdam`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isPremium: true,
  isFavorite: false,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 111,
  },
}, {
  id: `2020`,
  city: {
    location: {
      lat: 52.3909553943508,
      long: 4.85309666406198,
      zoom: 13,
    },
    name: `Paris`,
  },
  location: {
    lat: 52.3909553943508,
    long: 4.85309666406198,
    zoom: 13,
  },
  pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  preview: `img/apartment-01.jpg`,
  title: `Wood and stone place`,
  type: `room`,
  price: 999,
  isFavorite: false,
  isPremium: true,
  rating: 4.4,
  description: ``,
  bedrooms: 2,
  guests: 4,
  features: [],
  owner: {
    avatar: `https://api.adorable.io/avatars/128`,
    name: ``,
    isSuper: true,
    id: 222,
  },
}];

export const offersRaw = [{
  "id": 1010,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Amsterdam`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_premium": true,
  "is_favorite": false,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 111,
  },
}, {
  "id": 2020,
  "city": {
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 13,
    },
    "name": `Paris`,
  },
  "location": {
    "latitude": 52.3909553943508,
    "longitude": 4.85309666406198,
    "zoom": 13,
  },
  "images": [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
  "preview_image": `img/apartment-01.jpg`,
  "title": `Wood and stone place`,
  "type": `room`,
  "price": 999,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4.4,
  "description": ``,
  "bedrooms": 2,
  "max_adults": 4,
  "goods": [],
  "host": {
    "avatar_url": `https://api.adorable.io/avatars/128`,
    "name": ``,
    "is_pro": true,
    "id": 222,
  },
}];

export const reviews = [{
  id: 1110,
  avatar: `https://api.adorable.io/avatars/128`,
  name: `Angelina`,
  rating: 4.4,
  date: new Date(2020, 10, 5, 10),
  text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
}];

export const authInfo = {
  avatar: `https://api.adorable.io/avatars/128`,
  email: `mail@mail.ru`,
  id: 1,
  isSuper: true,
  name: `Angelina`,
};
