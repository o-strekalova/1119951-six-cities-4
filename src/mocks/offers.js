const OFFERS_TITLES = [`Beautiful & luxurious apartment at great location`, `Wood and stone place`, `Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`];
const OFFERS_TYPES = [`apartment`, `room`, `house`, `hotel`];
const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const COORDINATES = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];
const OFFERS_COUNT = 4;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const getRandomArrayItem = (array) => {
  const randomIndex = getRandomIntegerNumber(0, array.length);
  return array[randomIndex];
};

const generateOffers = () => {
  const offers = [];

  for (let i = 0; i < OFFERS_COUNT; i++) {
    offers.push({
      id: i,
      coords: COORDINATES[i],
      pictures: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
      title: getRandomArrayItem(OFFERS_TITLES),
      type: getRandomArrayItem(OFFERS_TYPES),
      price: getRandomIntegerNumber(50, 1000),
      isPremium: Math.random() > 0.5,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
      bedrooms: getRandomIntegerNumber(1, 10),
      guests: getRandomIntegerNumber(1, 20),
      features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
      owner: {
        avatar: AVATAR_URL,
        name: `Angelina`,
        isSuper: Math.random() > 0.5,
      },
    });
  }

  return offers;
};

export const offers = generateOffers();
