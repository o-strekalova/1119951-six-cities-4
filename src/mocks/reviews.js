const AVATAR_URL = `https://api.adorable.io/avatars/128`;
const REVIEWS_COUNT = 11;

const getRandomIntegerNumber = (min, max) => {
  return min + Math.floor(Math.random() * (max - min));
};

const generateReviews = () => {
  const reviews = [];

  for (let i = 0; i < REVIEWS_COUNT; i++) {
    reviews.push({
      id: i,
      avatar: AVATAR_URL,
      name: `Angelina`,
      rating: getRandomIntegerNumber(0, 4) + Math.round(Math.random() * 10) / 10,
      date: new Date(),
      text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.`,
    });
  }

  return reviews;
};

export const reviews = generateReviews();
