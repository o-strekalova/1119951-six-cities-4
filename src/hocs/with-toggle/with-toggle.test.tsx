import React from "react";
import renderer from "react-test-renderer";
import withToggle from "./with-toggle";
import {offersAll} from "../../components/mocks";

const MockComponent = () => {
  return <div/>;
};

const MockComponentWrapped = withToggle(MockComponent);

it(`withToggle is rendered correctly for Card`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      offer={offersAll[0]}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`withToggle is rendered correctly for Property`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      offer={offersAll[1]}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
