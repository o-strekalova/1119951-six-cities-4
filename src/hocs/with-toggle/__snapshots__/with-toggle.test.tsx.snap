// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withToggle is rendered correctly for Card 1`] = `<div />`;

exports[`withToggle is rendered correctly for Property 1`] = `<div />`;
